var/spack/repos/builtin/packages/fenics2019/package.py

```diff
@@ -0,0 +1,144 @@
+    depends_on('hdf5+hl+fortran', when='+hdf5+petsc')
+    depends_on('hdf5+hl', when='+hdf5~petsc')
+    depends_on('metis+real64', when='+parmetis')
+    depends_on('parmetis', when='+parmetis')
+    depends_on('scotch~metis', when='+scotch~mpi')
+    depends_on('scotch+mpi~metis', when='+scotch+mpi')
+    depends_on('petsc', when='+petsc')
+    depends_on('slepc', when='+slepc')
+    depends_on('py-petsc4py@3.6:', when='+petsc+python')
+    depends_on('trilinos', when='+trilinos')
+    depends_on('vtk', when='+vtk')
+    depends_on('suite-sparse@4.5.6', when='+suite-sparse')
+    depends_on('qt', when='+qt')
+
+    depends_on('py-pybind11', type=('build', 'run'))
+    depends_on('cmake@3.17.3:', type='build')
+
+    depends_on('py-setuptools', type='build', when='+python')
+    depends_on('py-pkgconfig', type='run', when='+python')
+    depends_on('py-sphinx@1.0.1:', when='+doc', type='build')
+
+    def cmake_is_on(self, option):
+        return 'ON' if option in self.spec else 'OFF'
+
+    def cmake_args(self):
+        spec = self.spec
+
+        opts = [
+            '-DDOLFIN_ENABLE_DOCS:BOOL={0}'.format(self.cmake_is_on('+doc')),
+            '-DBUILD_SHARED_LIBS:BOOL={0}'.format(self.cmake_is_on('+shared')),
+            '-DDOLFIN_SKIP_BUILD_TESTS:BOOL=ON',
+            '-DDOLFIN_ENABLE_OPENMP:BOOL={0}'.format(self.cmake_is_on('+openmp')),
+            '-DDOLFIN_ENABLE_CHOLMOD:BOOL={0}'.format(
+                self.cmake_is_on('suite-sparse')),
+            '-DDOLFIN_ENABLE_HDF5:BOOL={0}'.format(self.cmake_is_on('hdf5')),
+            '-DDOLFIN_ENABLE_MPI:BOOL={0}'.format(self.cmake_is_on('mpi')),
+            '-DDOLFIN_ENABLE_PARMETIS:BOOL={0}'.format(
+                self.cmake_is_on('parmetis')),
+            '-DDOLFIN_ENABLE_PASTIX:BOOL={0}'.format(self.cmake_is_on('pastix')),
+            '-DDOLFIN_ENABLE_PETSC:BOOL={0}'.format(self.cmake_is_on('petsc')),
+            '-DDOLFIN_ENABLE_PETSC4PY:BOOL={0}'.format(
+                self.cmake_is_on('py-petsc4py')),
+            '-DDOLFIN_ENABLE_PYTHON:BOOL={0}'.format(self.cmake_is_on('python')),
+            '-DDOLFIN_ENABLE_QT:BOOL={0}'.format(self.cmake_is_on('qt')),
+            '-DDOLFIN_ENABLE_SCOTCH:BOOL={0}'.format(self.cmake_is_on('scotch')),
+            '-DDOLFIN_ENABLE_SLEPC:BOOL={0}'.format(self.cmake_is_on('slepc')),
+            '-DDOLFIN_ENABLE_SLEPC4PY:BOOL={0}'.format(
+                self.cmake_is_on('py-slepc4py')),
+            '-DDOLFIN_ENABLE_SPHINX:BOOL={0}'.format(self.cmake_is_on('py-sphinx')),
+            '-DDOLFIN_ENABLE_TRILINOS:BOOL={0}'.format(
+                self.cmake_is_on('trilinos')),
+            '-DDOLFIN_ENABLE_UMFPACK:BOOL={0}'.format(
+                self.cmake_is_on('suite-sparse')),
+            '-DDOLFIN_ENABLE_VTK:BOOL={0}'.format(self.cmake_is_on('vtk')),
+            '-DDOLFIN_ENABLE_ZLIB:BOOL={0}'.format(self.cmake_is_on('zlib')),
+        ]
+        os.environ['PETSC_DIR'] = spec['petsc'].prefix
```

Comment:
Is this necessary, or can these be found through CMAKE_PREFIX_PATH too, which spack sets by default for deps?

Reply:
agreed, set of os.environ will be removed